src/pages/new-chat/new-chat.html

```diff
@@ -10,9 +10,15 @@
 
 <ion-content class="new-chat">
   <ion-list class="users">
-    <button ion-item *ngFor="let user of users | async" class="user" (click)="addChat(user)">
+    <button ion-item *ngFor="let user of users | async | search:searchPattern:'profile.name'" class="user" (click)="addChat(user)">
       <img class="user-picture" [src]="user.profile.picture">
       <h2 class="user-name">{{user.profile.name}}</h2>
     </button>
   </ion-list>
 </ion-content>
+
+<ion-footer>
+  <ion-toolbar color="whatsapp" position="bottom">
+    <ion-searchbar color="whatsapp" [(ngModel)]="searchPattern"></ion-searchbar>
```

Comment:
I'd prefer to just do something like:

```html
<ion-searchbar (ionInput)="searchPattern = $event.target.value" />
```

rather than the solution you provided, because this is much simpler and it is based on a one-way data-binding.